analyzer/workers/channel.c

```diff
@@ -43,41 +44,25 @@
 SUINLINE void
 suscan_analyzer_read_start(suscan_analyzer_t *analyzer)
 {
-#ifdef __linux__
-  clock_gettime(CLOCK_MONOTONIC_COARSE, &analyzer->read_start);
-#else
-  clock_gettime(CLOCK_MONOTONIC, &analyzer->read_start);
-#endif
+  analyzer->read_start = suscan_gettime_coarse();
 }
 
 SUINLINE void
 suscan_analyzer_process_start(suscan_analyzer_t *analyzer)
 {
-#ifdef __linux__
-  clock_gettime(CLOCK_MONOTONIC_COARSE, &analyzer->process_start);
-#else
-  clock_gettime(CLOCK_MONOTONIC, &analyzer->process_start);
-#endif
+  analyzer->process_start = suscan_gettime_coarse();
 }
 
 SUINLINE void
 suscan_analyzer_process_end(suscan_analyzer_t *analyzer)
 {
-  struct timespec sub;
   uint64_t total, cpu;
 
-#ifdef __linux__
-  clock_gettime(CLOCK_MONOTONIC_COARSE, &analyzer->process_end);
-#else
-  clock_gettime(CLOCK_MONOTONIC, &analyzer->process_end);
-#endif
+  analyzer->process_end = suscan_gettime_coarse();
 
-  if (analyzer->read_start.tv_sec > 0) {
```

Comment:
I'm not sure what this check is supposed to do?  Is it supposed to check that timestamp != 0? Or that it is >= 1sec? I *think* it is != 0, in which case the replacement expression can be simplified.

Reply:
The former. We are comparing read_start with an "invalid time" constant. Using != makes way more sense here.

Reply:
Thought so.  Code already changed.